src/DevYeah.LMS/DevYeah.LMS.Business/EmailClient.cs

```diff
@@ -19,31 +20,58 @@ public EmailClient(IOptions<EmailSettings> emailSettings)
         public void SendEmail(string email, string subject, string content)
         {
             if (string.IsNullOrWhiteSpace(email)) throw new ArgumentNullException(nameof(email));
+            if (string.IsNullOrWhiteSpace(subject)) throw new ArgumentNullException(nameof(subject));
             if (string.IsNullOrWhiteSpace(content)) throw new ArgumentNullException(nameof(content));
 
             var message = WrappingEmail(email, subject, content);
+            RetryAction(() => Sending(message), _emailSettings.MaxRetryCount);
+        }
+
+        private void Sending(MimeMessage message)
```

Comment:
Normally a function name should be an action, so `Sending` should be just `Send`. It's fine, it's compile time polymorphism, aka overloads.